sdk/identity/azure-identity/src/main/java/com/azure/identity/IdentityLogOptions.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.identity;
+
+import com.azure.core.annotation.Fluent;
+
+/**
+ * Represents the Identity Log options used to configure logging on the Token Credentials.
+ */
+@Fluent
+public final class IdentityLogOptions {
+    private boolean allowLoggingAccountIdentifiers;
+
+    /**
+     * Sets the flag indicating whether client side account identifier logs should be logged or not.
+     * The default value is false.
+     * <p>
+     * The Account Identifier logs can contain sensitive information and should be enabled on protected machines only.
+     * <p>
+     *
+     * @param allowLoggingAccountIdentifiers The flag indicating if client side account identifier logging should be
+     * enabled or not.
+     * @return The updated IdentityClientOptions object.
+     */
+    public IdentityLogOptions setAllowLoggingAccountIdentifiers(boolean allowLoggingAccountIdentifiers) {
+        this.allowLoggingAccountIdentifiers = allowLoggingAccountIdentifiers;
+        return this;
+    }
+
+    /**
+     * Gets the status indicating if Account Identifier Logging is allowed or not.
```

Comment:
In the JavaDocs, we should also call out that logging level INFO should be enabled for this to take effect. Also, if possible, list out the account identifiers or at least the most common fields that are considered account identifiers.

Reply:
Updated docs of setter.